### ISSUE TITLE
[Bug] Theme sometimes do not follow system dark mode on custom or auto schedule

### ISSUE THREAD

**Issue by BrayanDSO** [ankidroid/Anki-Android#10619]
###### Reproduction Steps

1. Use dark mode on custom or auto schedule
2. Disable/Enable dark mode manually

https://user-images.githubusercontent.com/69634269/160028344-bd1ac80b-c9dd-4a93-a8cc-b993d3fd1d16.mp4

###### Expected Result

Theme follow system correctly

###### Actual Result

Theme do not follow system

###### Debug info
```
AnkiDroid Version = 2.16alpha53

Android Version = 11

Manufacturer = samsung

Model = SM-N770F

Hardware = exynos9810

Webview User Agent = Mozilla/5.0 (Linux; Android 11; SM-N770F Build/RP1A.200720.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/99.0.4844.73 Mobile Safari/537.36

ACRA UUID = 037bbc86-38a4-4ad7-9c42-05870e9b6086

Scheduler = std2

Crash Reports Enabled = true

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [X] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] I have searched for similar existing issues here and on the user forum
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by BrayanDSO** [comment]
The attached video shows the behavior is kind of hard to reproduce, since even with custom scheduling it still need a manual dark mode trigger to show the bug.

I've assessed the problem and it looks like is a lack of handling of the auto/custom themes, which is not straightforward since two libraries on Android handle them differently (https://developer.android.com/reference/android/app/UiModeManager and https://developer.android.com/reference/androidx/appcompat/app/AppCompatDelegate)

I'll fix it when I get the time.

**Comment by mikehardy** [comment]
Not sure it's an either/or or a "both" type of choice for solution, but if it's either/or then always prefer androidx because that one is library upgradable and the other is tied to platform upgrade cycle (so basically never updates)

**Comment by BrayanDSO** [comment]
Reverted. Patched by #11555
